src/actions/transactions.js

```diff
@@ -311,3 +267,33 @@ export const sent = ({
       dispatch(passphraseUsed(passphrase));
     }
   };
+
+/**
+ * This action is used in account middleware to update transactions when new block was created.
+ * TODO this action should be likely removed when account middleware is cleaned up in
+ * https://github.com/LiskHQ/lisk-hub/issues/2021
+ *
+ * @param {Object} param.transactions - the transaction object with pending and confirmed list
+ * @param {Object} param.account - an object with active account
+ * @param {Boolean} windowFocus - flag that says if the Lisk Hub window is focused
+ */
+export const updateTransactionsIfNeeded = ({ transactions, account }, windowFocus) =>
```

Comment:
this function is completely unnecessary and it has a very inappropriate name and it needs to be removed. Move the condition to the account then you `transactionsUpdated` inside the account action or middleware.

Reply:
Done